amps/ags/rm-community/rm-community-repo/unit-test/java/org/alfresco/module/org_alfresco_module_rm/hold/HoldServiceImplUnitTest.java

```diff
@@ -306,6 +307,72 @@ public void setHoldReason()
         verify(mockedNodeService).setProperty(hold, PROP_HOLD_REASON, HOLD_REASON);
     }
 
+    @Test
+    public void setHoldDeletionReason()
+    {
+        // setup node service interactions
+        when(mockedNodeService.exists(hold))
+            .thenReturn(false)
+            .thenReturn(true)
+            .thenReturn(true);
+
+        // node does not exist
+        holdService.setHoldDeletionReason(hold, HOLD_DELETION_REASON);
+        verify(mockedNodeService, never()).setProperty(hold, PROP_HOLD_DELETION_REASON, HOLD_DELETION_REASON);
+
+        // node isn't a hold
+        holdService.setHoldDeletionReason(recordFolder, HOLD_DELETION_REASON);
+        verify(mockedNodeService, never()).setProperty(hold, PROP_HOLD_DELETION_REASON, HOLD_DELETION_REASON);
+
+        // set hold deletion reason
+        holdService.setHoldDeletionReason(hold, HOLD_DELETION_REASON);
+        verify(mockedNodeService).setProperty(hold, PROP_HOLD_DELETION_REASON, HOLD_DELETION_REASON);
```

Comment:
Despite the format of the existing unit tests I would still prefer having a single call to the method under test per test. It's not a problem to stick with this format (it's clear enough), but as a general point please don't feel obliged to be consistent if there's a better approach we could use and if it doesn't incur extra cost. Some of the AGS code was written a long time ago.